yb-voyager/src/queryissue/unsupported_dml_constructs.go

```diff
@@ -111,3 +111,93 @@ func (d *XmlExprDetector) Detect(msg protoreflect.Message) ([]string, error) {
 	}
 	return nil, nil
 }
+
+type RangeTableFunDetector struct{}
+
+func NewRangeTableFuncDetector() *RangeTableFunDetector {
+	return &RangeTableFunDetector{}
+}
+
+// Detect checks if a RangeTableFunc node is present for a XMLTABLE() function
+func (d *RangeTableFunDetector) Detect(msg protoreflect.Message) ([]string, error) {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_RANGETABLEFUNC_NODE {
+		if isXMLTable(msg) {
+			return []string{XML_FUNCTIONS}, nil
+		}
+	}
+	return nil, nil
+}
+
+/*
+XMLTABLE transforms XML data into relational table format, making it easier to query XML structures.
+Detection in RangeTableFunc Node:
+- docexpr: Refers to the XML data source, usually a column storing XML.
+- rowexpr: XPath expression (starting with '/' or '//') defining the rows in the XML.
+- columns: Specifies the data extraction from XML into relational columns.
+
+Example: Converting XML data about books into a table:
+SQL Query:
+
+	SELECT x.*
+	FROM XMLTABLE(
+		'/bookstore/book'
+		PASSING xml_column
+		COLUMNS
+			title TEXT PATH 'title',
+			author TEXT PATH 'author'
+	) AS x;
+
+Here, 'docexpr' points to 'xml_column' containing XML data, 'rowexpr' selects each 'book' node, and 'columns' extract 'title' and 'author' from each book.
+Hence Presence of XPath in 'rowexpr' and structured 'columns' typically indicates XMLTABLE usage.
+
+Function to detect if a RangeTableFunc node represents XMLTABLE()
+*/
+func isXMLTable(rangeTableFunc protoreflect.Message) bool {
+	log.Infof("checking if range table func node is for XMLTABLE()")
+	// Check for 'docexpr' field
+	docexprField := rangeTableFunc.Descriptor().Fields().ByName("docexpr")
+	if docexprField == nil {
+		return false
+	}
+	docexprNode := rangeTableFunc.Get(docexprField).Message()
```

Comment:
I'm not entirely sure, but from the docs it seems like this docexpr node can be NULL? 
https://www.postgresql.org/docs/current/functions-xml.html#:~:text=No%20rows%20will%20be%20produced%20if%20the%20document_expression%20is%20null%2C%20nor%20if%20the%20row_expression%20produces%20an%20empty%20node%2Dset%20or%20any%20value%20other%20than%20a%20node%2Dset.

Reply:
Ah, understood!